opaque_keys/edx/locator.py

```diff
@@ -1620,3 +1621,59 @@ def html_id(self) -> str:
         # HTML5 allows ID values to contain any characters at all other than spaces.
         # These key types don't allow spaces either, so no transform is needed.
         return str(self)
+
+
+class LibraryCollectionLocator(CheckFieldMixin, LibraryCollectionKey):
+    """
+    When serialized, these keys look like:
+        lib-collection:org:lib:collection-id
+    """
+    CANONICAL_NAMESPACE = 'lib-collection'
+    KEY_FIELDS = ('org', 'lib', 'usage_id')
+    org: str
+    lib: str
+    lib_key: LibraryLocatorV2
+    usage_id: str
```

Comment:
I'm not sure if `usage_id` is the best name for this slug field. `usage_id` usually refers to a whole `UsageKey`, not a short slug, which is usually called `block_id`. So this short slug is not a `usage_id`. I think it would be better called a `collection_id` or `slug` or `code`, but probably `collection_id`

Reply:
Updated: https://github.com/openedx/opaque-keys/pull/335/commits/168c373f4adafe7c81f707baef3a971f7404acd7